library/src/main/kotlin/me/thanel/swipeactionview/SwipeActionView.kt

```diff
@@ -478,9 +479,16 @@ class SwipeActionView : FrameLayout {
      * movement animation. (Defaults to 0)
      */
     @JvmOverloads
-    fun animateToOriginalPosition(startDelay: Long = 0) {
+    fun animateToOriginalPosition(startDelay: Long = 0, swipeDirection: SwipeDirection? = null) {
```

Comment:
This is a bit problematic. Since that is a public function users can invoke it, but then they might incorrectly pass the wrong swipe direction. Technically it would be their error but it would be nice if we could protect against this.

We should make this version of the function with 2 parameters private and add a second version without this new parameter. Then I can see 2 options: either always pass null from that public function or somehow track which direction was swiped recently and pass that. I'm in favor of the second option as this will make new callbacks always functional but I'm not sure how easy it would be to implement.

Reply:
It might not be clear enough from what I wrote, but my intention was to only trigger the callback of completion if the swipe action was completed. So if you drag a little bit, but don't "complete" the action triggering onSwipedRight/onSwipedLeft, the swipe complete callback would not be triggered either. So even if I could determine which direction it was in every scenario, I only really want to trigger the callback when the user finished a proper swipe. 

Maybe I can pass in a callback instead of swipe direction and make that version private?

Reply:
I see, in that case, it makes sense to pass null. 

Here I'm talking about a different situation. The library allows you to return false from onSwipedLeft/Right functions. If you do that you then to have manually call this function to move view back to the original position. This to me feels like a candidate for the new swipe completed callbacks and which could benefit from automatic direction detection.

But perhaps in that “manual” flow, these new callbacks aren't useful? I think I'll just have to run this on a device to see what is best as I'm not sure right now.

Reply:
Mentioned callback might be a good idea. That way if someone wanted confirmation in manual animation they would pass it here.

Reply:
I pushed those changes, let me know what you think.

Reply:
Looks good, just a minor suggestion to avoid creating Runnable.